### ISSUE TITLE
[Bug]: PatchCollection fails to keep hatch from Patch

### ISSUE THREAD

**Issue by GYHHAHA** [matplotlib/matplotlib#22654]
### Bug summary

When using add_collection to add Patches, hatch attribute isn't able to kept, whereas add_patch is fine.

### Code for reproduction

```python
from matplotlib.patches import Rectangle
import matplotlib.pyplot as plt
from matplotlib.collections import PatchCollection
fig, ax = plt.subplots(figsize=(4,4))
patches = [
    Rectangle((0, 0), 0.2, .2, facecolor="orange", hatch="/")
]
collected_patches = PatchCollection(patches, match_original=True)
ax.add_collection(collected_patches)
ax.add_patch(Rectangle((0.8, 0.8), 0.2,0.2, facecolor="skyblue", hatch="/"))
```

### Actual outcome

![output](https://user-images.githubusercontent.com/41546976/158526503-39a93e32-366f-4d4c-8285-a9f506a7fa40.png)

### Expected outcome

![output2](https://user-images.githubusercontent.com/41546976/158526515-ebbedbde-de67-4692-8ddf-2a1052f796a7.png)

### Matplotlib Version

3.5.1

**Comment by Impaler343** [comment]
Hey, I was looking to solve this issue, and according to this [PR](https://github.com/matplotlib/matplotlib/pull/22814), we are trying to use only one hatch for all patches based on the first patch, but what if we needed to use different hatches for different patches? This would not be possible. Could someone elaborate on why we shouldn't be implementing it in the second way?

**Comment by timhoffm** [comment]
@Impaler343 You are correct. That PR is too simple (well it would have been a step in the right direction). Fundamentally, we'd want individual hatches.
